### PR TITLE
feat(python/rust): add support std & var for `duration` series

### PR DESCRIPTION
**Pull request by ion-elgreco**
Added support for std and var for duration series. It goes from int64 -> float64 -> int64.

- closes https://github.com/pola-rs/polars/issues/12294

### CODE REVIEW

File: crates/polars-core/src/chunked_array/logical/duration.rs [hunk 1]
```diff
@@ -11,6 +11,26 @@ impl Int64Chunked {
     }
 }
 
+impl Float64Chunked {
```
Comment (ritchie46) [inline crates/polars-core/src/chunked_array/logical/duration.rs:14]: `into_duration` make no sense for floating point arrays.

You should cast somewhere else and then call `into_duration`.

File: crates/polars-core/src/series/implementations/duration.rs [hunk 2]
```diff
@@ -407,6 +407,13 @@ impl SeriesTrait for SeriesWrap<DurationChunked> {
     fn min_as_series(&self) -> Series {
         self.0.min_as_series().into_duration(self.0.time_unit())
     }
+    fn std_as_series(&self, ddof: u8) -> Series {
+        self.0.std_as_series(ddof).into_duration(self.0.time_unit())
```
Comment (ritchie46) [inline crates/polars-core/src/series/implementations/duration.rs:411]: So here you must cast the result back to integers and then call `into_duration`.

Comment (ion-elgreco) [reply crates/polars-core/src/series/implementations/duration.rs]: Done! :)

**Comment by ritchie46** [comment]
Yeap, looks good. Thanks!

**Comment by mcrumiller** [comment]
@ritchie46 and @ion-elgreco can we reopen this? We need to add series support, not just expr support:

```python
import polars as pl
from datetime import timedelta

s_duration = pl.Series([timedelta(days=1), timedelta(days=2), timedelta(days=4)])

s_duration.std()  # does not work (returns null)
pl.DataFrame({"a": s}).select(pl.col("a").std())  # works
```

**Pull request merged**